src/library/scala/collection/View.scala

```diff
@@ -163,6 +163,64 @@ object View extends IterableFactory[View] {
     def iterator = partition.underlying.iterator.filter(x => partition.p(x) == cond)
   }
 
+  /** A view that splits an underlying collection into two views */
+  class MapSplit[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) {
```

Comment:
Needs a `@SerialVersionUID`

Reply:
Are you sure this class does need this annotation? As it doesn't extend a view... when I did add the annotation, I get this warning ```@SerialVersionUID has no effect on non-serializable classes
[warn]   class PartitionWith[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) ```